store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorageTest.java

```diff
@@ -65,491 +69,490 @@
-        final List<String> configAB = Arrays.asList(a.getDeserialisedConfig().getGraphId(), b.getDeserialisedConfig().getGraphId());
-        final List<String> configBA = Arrays.asList(b.getDeserialisedConfig().getGraphId(), a.getDeserialisedConfig().getGraphId());
+        graphStorage.put(Lists.newArrayList(graphSerialisableA, graphSerialisableB), auth1Access);
+        final List<String> configAB = Arrays.asList(GRAPH_ID_A, GRAPH_ID_B);
+        final List<String> configBA = Arrays.asList(GRAPH_ID_B, GRAPH_ID_A);
 
         // When
-        final Collection<Graph> graphsAB = graphStorage.get(authUser, configAB);
-        final Collection<Graph> graphsBA = graphStorage.get(authUser, configBA);
+        final Collection<Graph> graphsAB = graphStorage.get(authUser(), configAB);
+        final Collection<Graph> graphsBA = graphStorage.get(authUser(), configBA);
 
         // Then
         // A B
-        final Iterator<Graph> itrAB = graphsAB.iterator();
-        assertSame(a.getGraph(), itrAB.next());
-        assertSame(b.getGraph(), itrAB.next());
-        assertThat(itrAB).isExhausted();
+        assertThat(graphsAB).containsExactly(graphSerialisableA.getGraph(), graphSerialisableB.getGraph());
         // B A
-        final Iterator<Graph> itrBA = graphsBA.iterator();
-        assertSame(b.getGraph(), itrBA.next());
-        assertSame(a.getGraph(), itrBA.next());
-        assertThat(itrBA).isExhausted();
+        assertThat(graphsBA).containsExactly(graphSerialisableB.getGraph(), graphSerialisableA.getGraph());
     }
 
     @Test
     public void shouldNotAddGraphWhenLibraryThrowsExceptionDuringAdd() throws Exception {
         //given
         GraphLibrary mock = mock(GraphLibrary.class);
         String testMockException = "testMockException";
-        String graphId = a.getDeserialisedConfig().getGraphId();
         Mockito.doThrow(new RuntimeException(testMockException))
                 .when(mock)
-                .checkExisting(graphId, a.getDeserialisedSchema(), a.getDeserialisedProperties());
+                .checkExisting(GRAPH_ID_A, graphSerialisableA.getDeserialisedSchema(), graphSerialisableA.getDeserialisedProperties());
         graphStorage.setGraphLibrary(mock);
-        try {
-            graphStorage.put(a, access);
-            fail(EXCEPTION_EXPECTED);
-        } catch (final Exception e) {
-            assertTrue(e instanceof StorageException);
-            assertEquals(testMockException, e.getCause().getMessage());
-        }
-        try {
-            //when
-            graphStorage.get(testUser, Lists.newArrayList(graphId));
-            fail(EXCEPTION_EXPECTED);
-        } catch (final IllegalArgumentException e) {
-            //then
-            assertEquals(String.format(GRAPH_IDS_NOT_VISIBLE, Arrays.toString(new String[]{graphId})), e.getMessage());
-        }
+
+        final StorageException storageException = assertThrows(StorageException.class, () -> graphStorage.put(graphSerialisableA, auth1Access));
```

Comment:
You can use assertJ here and it's a bit [cleaner](https://github.com/gchq/Gaffer/blob/2bb2ee977049d39381ea314691161d9c08615bfa/store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java#L278)

Reply:
Yes. I hadn't seen that way of writing it until much later. I was copying other approached to excepting handling.

Reply:
Are you going to update these to use assertJ? Probably don't have to but would be preferred

Reply:
I'm looking at it today. It doesn't follow the //given //when //then as closely.
I think the two are very similar, but `assertThatExceptionOfType` does look nicer.